RuntimeWarning: invalid value encountered in divide in my local dev environment

Oddly enough it looks like the test passes in the CI, but is failing in my local dev environment.

Locally I get:

```bash
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <Task 'truediv-633b519b183f0ae6f0e12d1c420b78e0_.0_.1_.2' truediv(...)>
values = {'__dask_blockwise__1': array([[[0.        , 0.        , 0.        , ..., 0.        ,
         0.        , 0.        ]...
        [0.        , 0.        , 0.        , ..., 0.        ,
         0.        , 0.        ]]], shape=(72, 14, 24))}

    def __call__(self, values=()):
        self._verify_values(values)
        new_argspec = tuple(
            a({k: values[k] for k in a.dependencies}) if isinstance(a, GraphNode) else a
            for a in self.args
        )
        if self.kwargs:
            kwargs = {
                k: (
                    kw({k: values[k] for k in kw.dependencies})
                    if isinstance(kw, GraphNode)
                    else kw
                )
                for k, kw in self.kwargs.items()
            }
            return self.func(*new_argspec, **kwargs)
>       return self.func(*new_argspec)
E       RuntimeWarning: invalid value encountered in divide

new_argspec = (array([[[0.        , 0.        , 0.        , ..., 0.        ,
         0.        , 0.        ],
        [0.        , ...
        [0.        , 0.        , 0.        , ..., 0.        ,
         0.        , 0.        ]]], shape=(72, 14, 24)))
self       = <Task 'truediv-633b519b183f0ae6f0e12d1c420b78e0_.0_.1_.2' truediv(...)>
values     = {'__dask_blockwise__1': array([[[0.        , 0.        , 0.        , ..., 0.        ,
         0.        , 0.        ]...
        [0.        , 0.        , 0.        , ..., 0.        ,
         0.        , 0.        ]]], shape=(72, 14, 24))}

../miniconda3/envs/pyrte-311/lib/python3.11/site-packages/dask/_task_spec.py:741: RuntimeWarning

---------- coverage: platform darwin, python 3.11.8-final-0 ----------
Name                                       Stmts   Miss  Cover   Missing
------------------------------------------------------------------------
pyrte_rrtmgp/cli.py                           49     49     0%   3-106
pyrte_rrtmgp/data_validation.py               73      8    89%   88, 91, 104, 118, 138, 143, 167, 191
pyrte_rrtmgp/external_data_helpers.py         40     40     0%   3-110
pyrte_rrtmgp/kernels/rte.py                   89     37    58%   168-190, 217-232, 339-350, 369-381, 400-412, 437-452, 509-523, 546-560, 664-677
pyrte_rrtmgp/rrtmgp_cloud_optics.py          124     15    88%   46, 51, 245-246, 324-383, 409, 437, 461, 552
pyrte_rrtmgp/rrtmgp_data.py                   39     10    74%   28, 32, 37, 70-79
pyrte_rrtmgp/rrtmgp_gas_optics.py            315     16    95%   66, 71, 121-131, 159, 227, 241, 713, 808, 810, 814
pyrte_rrtmgp/rte_solver.py                   104      5    95%   88, 146, 380, 385-386
pyrte_rrtmgp/tests/test_cld_sw_solver.py      59      4    93%   156-162
pyrte_rrtmgp/tests/test_lw_solver.py          49      1    98%   117
pyrte_rrtmgp/utils.py                         91      9    90%   54-63, 93, 95, 97
------------------------------------------------------------------------
TOTAL                                       1595    194    88%

10 files skipped due to complete coverage.

=========================== short test summary info ============================
FAILED pyrte_rrtmgp/tests/test_cld_sw_solver.py::test_sw_solver_with_clouds_dask
======================== 1 failed, 51 passed in 18.98s =========================
```

_Originally posted by @brendancol in https://github.com/earth-system-radiation/pyRTE-RRTMGP/issues/187#issuecomment-2779105685_

--- a/pyrte_rrtmgp/rrtmgp_cloud_optics.py
+++ b/pyrte_rrtmgp/rrtmgp_cloud_optics.py
@@ -24,6 +24,7 @@ from pyrte_rrtmgp.kernels.rte import (
     increment_2stream_by_2stream,
 )
 from pyrte_rrtmgp.rrtmgp_data import download_rrtmgp_data
+from pyrte_rrtmgp.utils import safer_divide
 
 
 def load_cloud_optics(
@@ -235,9 +236,26 @@ class CloudOpticsAccessor:
             taussa = ltaussa + itaussa
             taussag = ltaussag + itaussag
 
-            # Calculate derived quantities
-            ssa = xr.where(tau > np.finfo(float).eps, taussa / tau, 0.0)
-            g = xr.where(taussa > np.finfo(float).eps, taussag / taussa, 0.0)
+            # Apply the function with dask awareness.
+            ssa = xr.apply_ufunc(
+                safer_divide,
+                taussa,
+                tau,
+                dask="allowed",  # Allows lazy evaluation with dask arrays.
+                output_dtypes=[
+                    taussa.dtype
+                ],  # Ensure the output data type is correctly specified.
+            )
+
+            g = xr.apply_ufunc(
+                safer_divide,
+                taussag,
+                taussa,
+                dask="allowed",  # Allows lazy evaluation with dask arrays.
+                output_dtypes=[
+                    taussa.dtype
+                ],  # Ensure the output data type is correctly specified.
+            )
 
             props = xr.Dataset({"tau": tau, "ssa": ssa, "g": g})
 

--- a/pyrte_rrtmgp/utils.py
+++ b/pyrte_rrtmgp/utils.py
@@ -1,6 +1,7 @@
 """Utility functions for pyRTE-RRTMGP."""
 
 import os
+from typing import Any
 
 import numpy as np
 import xarray as xr
@@ -317,3 +318,8 @@ def load_rrtmgp_file(file: AllSkyExampleFiles | RFMIPExampleFiles) -> xr.Dataset
     rte_rrtmgp_dir = download_rrtmgp_data()
     ref_path = os.path.join(rte_rrtmgp_dir, file.value)
     return xr.load_dataset(ref_path, decode_cf=False)
+
+
+def safer_divide(a: Any, b: Any) -> Any:
+    """Safer np.divide util func."""
+    return np.divide(a, b, out=np.zeros_like(a), where=b > np.finfo(float).eps)

